### PR TITLE
Match orders in same block as they are placed

### PR DESCRIPTION
**Pull request by shubhamgoyal42**
## How this works
In commitNewWork, when committing each transactions, we check whether the order(s) in the transaction match with any existing orders, and matching transactions are included in the same block.

## How this was tested
On local

### CODE REVIEW

**Comment by shubhamgoyal42** [comment]
@atvanguard @debaghtk please help with naming on new structs, files and functions

File: .avalanche-cli.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "node-config": {
     "log-level": "info"
-  }
-}
+  },
+  "singlenodeenabled": false
```
Comment (debaghtk) [inline .avalanche-cli.json:5]: [nit] needs hyphenation like other values

File: miner/orderbook_checker.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import (
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/core/state"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type OrderBookChecker interface {
```
Comment (debaghtk) [inline miner/orderbook_checker.go:11]: nit - OrderbookChecker

File: .avalanche-cli.json (hunk 1 above)
Comment (shubhamgoyal42) [reply .avalanche-cli.json]: this name is interpreter by avalanche cli, we can't control it

File: .avalanche-cli.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "node-config": {
     "log-level": "info"
-  }
-}
+  },
+  "SingleNodeEnabled": false
```
[the remark this answers is not in the document]
Comment (shubhamgoyal42) [reply .avalanche-cli.json:5]: when it's true it means that the local network will run with 1 node instead of 5 nodes

**Pull request merged**